tests/src/guardfailure.rs

```diff
@@ -13,21 +13,21 @@ fn simple() {
     }
 
     #[interp_step]
-    fn interp_step(io: &mut IO) {
+    fn interp_step(io: &mut InterpCtx) {
         let x = guard(io.0);
         io.1 = x;
     }
 
-    let mut inputs = IO(std::hint::black_box(|i| i)(0), 0);
+    let mut ctx = InterpCtx(std::hint::black_box(|i| i)(0), 0);
     let th = start_tracing(TracingKind::HardwareTracing);
-    interp_step(&mut inputs);
+    interp_step(&mut ctx);
     let sir_trace = th.stop_tracing().unwrap();
     let ct = compile_trace(sir_trace).unwrap();
-    let mut args = IO(0, 0);
+    let mut args = InterpCtx(0, 0);
     assert!(unsafe { ct.execute(&mut args).is_null() });
     assert_eq!(args.1, 9);
-    // Execute trace with input that fails the guard.
-    let mut args = IO(3, 0);
+    // Execute the trace with the context that caused the guard to fail.
```

Comment:
The guard hasn't failed yet, but this context will make sure it does once we run it below. So the comment should be `Execute the trace with the context that causes the guard to fail.`

Reply:
I think it should then be "Execute the trace with a context that causes a guard to fail"?

Reply:
Agreed.

Reply:
Fixed in https://github.com/softdevteam/yk/pull/222/commits/903692a3b78345a26445326503b497dae085ac5c.